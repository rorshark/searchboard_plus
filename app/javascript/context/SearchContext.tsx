import React from 'react'
import { useSearch, SearchAPI, SearchParams } from '../hooks/useSearch'

export const defaultSearchApi: SearchAPI = {
  error: false,
  loading: true,
  customers: [],
  search: '',
  setSearch: (_) => {},
  companies: [],
  selectedCompany: null,
  setSelectedCompany: (_) => {},
  fetchCustomers: async () => {}
}

export const SearchContext: React.Context<SearchAPI> = React.createContext(defaultSearchApi)

export const SearchProvider: React.FC<{
  initialSearchParams: SearchParams
}> = ({ initialSearchParams, children }) => {
  const search = useSearch(initialSearchParams)

  return (
    <SearchContext.Provider value={search}>
      {children}
    </SearchContext.Provider>
  )
}
