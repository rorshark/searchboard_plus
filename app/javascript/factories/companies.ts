import { Factory } from 'fishery'
import Faker from 'faker'
import { Company } from '../types/companies'

export const company = Factory.define<Company>(() => ({
  companyName: Faker.company.companyName()
}))
