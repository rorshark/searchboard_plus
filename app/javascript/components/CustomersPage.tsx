import React from 'react'
import { SearchProvider } from '../context/SearchContext'
import CustomersTable from './CustomersTable'

export type BootstrapProps = {
  search_query: string
}

export const CustomersPage: React.FC<BootstrapProps> = () => (
  <SearchProvider>
    <CustomersTable />
  </SearchProvider>
)

export default CustomersPage
