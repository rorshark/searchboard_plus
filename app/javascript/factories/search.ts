import { Factory } from 'fishery'
import { SearchAPI, SearchParams } from '../hooks/useSearch'
import { DEFAULT_SEARCH_API } from '../context/SearchContext'

export const searchApi = Factory.define<SearchAPI>(() => DEFAULT_SEARCH_API)

export const searchParams = Factory.define<SearchParams>(() => ({
  search: '',
  filter_by_company_name: ''
}))
