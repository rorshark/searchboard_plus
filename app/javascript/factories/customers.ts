import { Factory } from 'fishery'
import Faker from 'faker'
import { Customer } from '../types/customers'
import { company } from './companies'

export const customer = Factory.define<Customer>(() => ({
  firstName: Faker.name.firstName(),
  lastName: Faker.name.lastName(),
  company: company.build()
}));
