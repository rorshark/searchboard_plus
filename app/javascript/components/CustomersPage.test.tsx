import React from 'react'
import { render, screen } from '@testing-library/react'
import { SearchContext } from '../context/SearchContext'
import { CustomersPage } from './CustomersPage'
import { searchApi } from '../factories/search'

test('it renders the customers page', () => {
  render(
    <SearchContext.Provider value={searchApi.build()}>
      <CustomersPage />
    </SearchContext.Provider>
  )

  const element = screen.getByTestId('customers-page')

  expect(element).toMatchSnapshot()
})
