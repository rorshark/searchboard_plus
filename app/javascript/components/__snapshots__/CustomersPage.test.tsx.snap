// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders the customers page 1`] = `
<div
  data-testid="customers-page"
>
  <h4
    class="MuiTypography-root MuiTypography-h4"
  >
    Customer Search
  </h4>
  <div
    data-testid="search-form"
  >
    <div
      class="MuiFormControl-root makeStyles-formControl-1"
    >
      <div
        class="MuiFormControl-root MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
        >
          Search
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            data-testid="search-input"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-4"
            >
              <span>
                Search
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      class="MuiFormControl-root makeStyles-formControl-1"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
        data-shrink="false"
        id="company-name"
      >
        Company Name
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      >
        <div
          aria-haspopup="listbox"
          aria-labelledby="company-name"
          class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
          role="button"
          tabindex="0"
        >
          <span>
            ​
          </span>
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput"
          data-testid="company-select"
          tabindex="-1"
          value=""
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
          style="padding-left: 8px;"
        >
          <legend
            class="PrivateNotchedOutline-legend-3"
            style="width: 0.01px;"
          >
            <span>
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
  <div
    data-testid="loading"
  >
    Loading...
  </div>
</div>
`;
